### PR TITLE
INT-3858: Doc File Inbound `scanner` and `filter`

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-3858

### CODE REVIEW

File: src/reference/asciidoc/file.adoc [hunk 1]
```diff
@@ -189,20 +189,26 @@ If this is a problem in your case you should call FileLocker.unlock(File file) y
...
-Any of these attributes set on the adapter are subsequently injected into the scanner.
-For this reason, if you need to provide a custom scanner and you have multiple file inbound adapters in the same application context, each adapter must be provided with its own instance of the scanner, either by declaring separate beans, or declaring `scope="prototype"` on the scanner bean so that the context will create a new instance for each use.
+IMPORTANT: It is important to understand that filters (including `patterns`, `regex`, `prevent-duplicates` etc) and `lockers`, are actually used by the `scanner`.
+Any of these attributes set on the adapter are subsequently injected into the `scanner`.
+For this reason, if you need to provide a custom `scanner` and you have multiple file inbound adapters in the same application context, each adapter must be provided with its own instance of the scanner, either by declaring separate beans, or declaring `scope="prototype"` on the scanner bean so that the context will create a new instance for each use.
```
Comment (garyrussell) [inline src/reference/asciidoc/file.adoc]: I guess this is not so important now, given that we prohibit the other properties so one adapter can't overwrite another's settings - however, it's still a good idea to use a different scanner for each instance, I guess.

Comment (artembilan) [inline src/reference/asciidoc/file.adoc]: Well, it depends on the `DirectoryScanner` implementation. For example our `DefaultDirectoryScanner` is stateless, so no reason to worry about sharing it between different adapters.
I found your  note as good catch.
Changing this paragraph to be consistent with existing state...

**Comment by artembilan** [comment]
Pushed

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/152f46719a8fbb7db5d24a5eb2c832f2c350d7b8

**Pull request closed without merging**